lib/logstash/filters/zeromq.rb

```diff
@@ -84,28 +88,44 @@ def close
 
   private
   def close
-    @logger.debug("0mq: closing socket.")
-    @poller.deregister(@zsocket, ZMQ::POLLIN)
-    @zsocket.close
+    @threads_sockets.each do |thread, zmq_info|
+      @logger.debug("0mq: closing socket.")
+      zmq_info.poller.deregister(zmq_info.zsocket, ZMQ::POLLIN)
+      zmq_info.zsocket.close
```

Comment:
Thinking through this,this *should* be OK so long as the pipeline has no messages in-flight when this runs. That is the case with the current pipeline, but I need to make a note of it here.